app/service/CharitiesKeyStoreService.scala

```diff
@@ -77,6 +86,11 @@ class CharitiesKeyStoreService @Inject()(cache: CharitiesShortLivedCache,
                       AuthorisedOfficialsStatusHelper.checkComplete(userAnswers) && AuthorisedOfficialsStatusHelper.validateDataFromOldService(userAnswers))
                     case _ => Success(userAnswers)
                   })
+                  .flatMap(userAnswers => userAnswers.get(Section8Page) match {
```

Comment:
it would be good to extract this as another private methd to reduce complexity of getCacheData method, applicable to earlier section flag as well

Reply:
will do